### PR TITLE
JSX transforms

### PR DESCRIPTION
**Pull request by danielberndt**
This PR adds support for the `<glamorous.Div>` transforms. It uses the `react-html-attributes` package for identifying all valid html attributes on an component and considers all other attributes as styling properties. 

- [x] gathers styling information into css property
- [x] uses `react-html-attributes` to distinguish html from css properties
- [x] add option flag to assume the presence of emotion-babel-plugin

### CODE REVIEW

**Comment by TejasQ** [comment]
Awesome! Let's review and merge this as soon as #2 is merged.

**Comment by TejasQ** [comment]
@danielberndt let's get this merged too 🚀

**Comment by danielberndt** [comment]
I just added the withBabelPlugin option flag for you to review :)

File: index.js [hunk 1]
```diff
@@ -39,16 +41,88 @@ module.exports = function(babel) {
     });
   };
 
+  // transform <glamorous.Div css={styles} width={100}/> to <div css={{...styles, width: 100}}/>
+  const transformJSXAttributes = ({tagName, jsxAttrs, withBabelPlugin, getCssFn, getCxFn}) => {
+    if (!jsxAttrs) return [];
+    const stylesArguments = [];
+    let classNameAttr = null;
+    let originalCssValue;
+    const newAttrs = jsxAttrs.filter(attr => {
```
Comment (TejasQ) [inline index.js]: Can we name `newAttrs` a bit more descriptively?

File: index.js [hunk 2]
```diff
@@ -101,30 +180,45 @@ module.exports = function(babel) {
...
 
         // use "styled" as new default import, only if there's no such variable in use yet
-        const newName = path.scope.hasBinding("styled")
-          ? path.scope.generateUidIdentifier("styled").name
-          : "styled";
+        const createUniqueIdentifier = name =>
+          path.scope.hasBinding(name) ? path.scope.generateUidIdentifier(name) : t.identifier(name);
         let newImports = [];
-        let useDefaultImport = false;
+        let emotionImports = {};
```
Comment (TejasQ) [inline index.js]: What does this do?

Comment (danielberndt) [reply index.js]: Depending on what features are needed, we potentially need to import the `default` export or `css` or `cx` from `react-emotion`.
These imports are added to the code [here](https://github.com/TejasQ/babel-plugin-glamorous-to-emotion/blob/899ec9772c3fdc78c2892e327bbec0bcedbee9b5/index.js#L238-L245)

File: index.js [hunk 3]
```diff
@@ -101,30 +180,45 @@ module.exports = function(babel) {
         }
 
         // use "styled" as new default import, only if there's no such variable in use yet
```
Comment (danielberndt) [inline index.js]: Ah, this comment is not up to date anymore. I'll fix it

Comment (TejasQ) [reply index.js]: @danielberndt looking forward to it

**Comment by danielberndt** [comment]
new update is in :)

File: index.js [hunk 4]
```diff
@@ -47,7 +47,7 @@ module.exports = function(babel) {
     const stylesArguments = [];
     let classNameAttr = null;
     let originalCssValue;
-    const newAttrs = jsxAttrs.filter(attr => {
+    const transformedJsxAttrs = jsxAttrs.filter(attr => {
```
Comment (TejasQ) [inline index.js]: Sorry to be a pain! But `transformedJsxAttrs` => transformed from what? A little 

```
/**
  * this does this
  */
```

would be helpful

Comment (TejasQ) [reply index.js]: Then we can merge it 🚀

File: README.md [hunk 5]
```diff
@@ -19,6 +19,21 @@ You'll need to use [`babel-codemod`](https://github.com/square/babel-codemod) to
...
+
+- **`withBabelPlugin=true`**
+
+  assumes that your emotion setup includes the [emotion-babel-plugin](https://github.com/emotion-js/emotion/tree/master/packages/babel-plugin-emotion). Without this option, `<glamorous.Div marginTop={5}/>` gets translated to `<div className={css({marginTop: 5})}>`.
+
+  If this option is enabled, it will be transformed to `<div css={{marginTop: 5}}>`.
+
+- **`preact=true`**
+
+  Uses `import styled from preact-emotion` instead of `import styled from react-emotion`
```
[the remark this answers is not in the document]
Comment (danielberndt) [reply README.md]: all done!

**Comment by TejasQ** [comment]
@danielberndt you rock so hard it's unreal. ⛰

**Comment by TejasQ** [comment]
I think you should [go over here](https://github.com/paypal/glamorous/issues/419) and tell everyone what's available now. 😉

**Comment by danielberndt** [comment]
Haha, good point! Done 👍

**Pull request merged**